freqtrade/optimize/backtesting.py

```diff
@@ -461,9 +459,19 @@ def backtest_one_strategy(self, strat: IStrategy, data: Dict[str, Any], timerang
             preprocessed[pair] = trim_dataframe(df, timerange,
                                                 startup_candles=self.required_startup)
         min_date, max_date = history.get_timerange(preprocessed)
+
+        # Overall data availability check. Nothing to do if we have no data at all.
         if min_date is NaT or max_date is NaT:
             raise OperationalException(
                 "No data left after adjusting for startup candles. ")
```

Comment:
I would almost think that this should happen in a different sequence.

move your "data removal" check to the top of this block (right before `get_timerange()`) - this is actually a better fix than the one i added before

check if ANY pair is left (if the dict is empty - no point in continuing) further.

then get the timerange - by which point, my previous "no data left after adjusting" fix might no longer be necessary - as the min/max time will only be NaT if one of the pairs has no data.